src/oauth2-client/controller/collection.ts

```diff
@@ -17,11 +20,65 @@ class ClientCollectionController extends Controller {
       }
     }
 
-    const clients = await findClientsByUser(user);
+    const clients = await findByUser(user);
     ctx.response.body = hal.collection(user, clients);
 
   }
 
+  async post(ctx: Context) {
+
+    const user = await userService.findById(ctx.state.params.id);
+    if (!await privilegeService.hasPrivilege(ctx, 'admin')) {
+      throw new Forbidden('Only users with the "admin" privilege can inspect OAuth2 clients that are not your own');
+    }
```

Comment:
I've seen this duplication of access control logic in a few of the controllers. It would be nice if there was a method per-controller that could be applied to _all_ methods in the controller. Then access control on that resource could be centralized and lower the possibility of forgetting about it.

This isn't at all blocking on this review; it's just an observation.

Reply:
Yeah totally agreed. Would be nice to have a controller-wide way to handle this. Was also thinking of doing something with decorators.

Reply:
Decorators would probably work, too. I've used them with other languages/frameworks. I keep forgetting that decorators are in draft proposal for JS.